crypto/ec/ecp_nistz256.c

```diff
@@ -1445,6 +1446,131 @@ static int ecp_nistz256_inv_mod_ord(const EC_GROUP *group, BIGNUM *r,
+        NULL;
+    EC_POINT *P = NULL;
+
+    if ((ctx = BN_CTX_new_ex(NULL)) == NULL) {
+        ERR_raise(ERR_LIB_EC, ERR_R_MALLOC_FAILURE);
+        goto err;
+    }
+    params -=seed_len;
+
+    if (!EC_GROUP_set_seed(group, params, seed_len)) {
+            ERR_raise(ERR_LIB_EC, ERR_R_EC_LIB);
+            goto err;
+    }
+    params += seed_len;
+
+    if ((p = BN_bin2bn(params + 0 * param_len, param_len, NULL)) == NULL
+        || (a = BN_bin2bn(params + 1 * param_len, param_len, NULL)) == NULL
+        || (b = BN_bin2bn(params + 2 * param_len, param_len, NULL)) == NULL) {
+        ERR_raise(ERR_LIB_EC, ERR_R_BN_LIB);
+        goto err;
+    }
+
+    /* Set up curve params and montgomery for field*/
+    /*  Start by setting up montgomery and one */
+    mont = BN_MONT_CTX_new();
+    if (mont == NULL)
+        goto err;
+
+    BN_MONT_CTX_init(mont);
+    mont->ri=256;
+    mont->n0[0] = 0x1;
+    mont->n0[1] = 0;
+    BN_copy(&mont->N, p);
+    BN_bin2bn(params+6*param_len, param_len, &mont->RR);
+
+    one = BN_new();
+    if (one == NULL){
+        ERR_raise(ERR_LIB_EC, ERR_R_BN_LIB);
+        goto err;
+    }
+    if (!BN_to_montgomery(one, BN_value_one(), mont, ctx)){
+        ERR_raise(ERR_LIB_EC, ERR_R_BN_LIB);
+        goto err;
+    }
+    group->field_data1 = mont;
+    mont = NULL;
+    group->field_data2 = one;
+    one = NULL;
+
+    if (!ossl_ec_GFp_simple_group_set_curve(group, p, a, b, ctx)) {
+         ERR_raise(ERR_LIB_EC, ERR_R_EC_LIB);
+        goto err;
+    }
+
+    if ((P = EC_POINT_new(group)) == NULL) {
+        ERR_raise(ERR_LIB_EC, ERR_R_EC_LIB);
+        goto err;
+    }
+
+    if ((x = BN_bin2bn(params + 3 * param_len, param_len, NULL)) == NULL
+        || (y = BN_bin2bn(params + 4 * param_len, param_len, NULL)) == NULL) {
+        ERR_raise(ERR_LIB_EC, ERR_R_BN_LIB);
+        goto err;
+    }
+    if (!EC_POINT_set_affine_coordinates(group, P, x, y, ctx)) {
+        ERR_raise(ERR_LIB_EC, ERR_R_EC_LIB);
+        goto err;
+    }
+    if ((order = BN_bin2bn(params + 5 * param_len, param_len, NULL)) == NULL
+        || !BN_set_word(x, (BN_ULONG)1)) { // cofactor is 1
+        ERR_raise(ERR_LIB_EC, ERR_R_BN_LIB);
+        goto err;
+    }
+
+    /* Set up generator and order and montgomery data */
+    group->generator = EC_POINT_new(group);
+    if (group->generator == NULL){
+         ERR_raise(ERR_LIB_EC, ERR_R_EC_LIB);
+         goto err;
+    }
+    if (!EC_POINT_copy(group->generator, P)) {
+        goto err;
+    }
+    if (!BN_copy(group->order, order)){
+        goto err;
+    }
+    if(!BN_copy(group->cofactor, BN_value_one())) {
+        goto err;
+    }
+    group->mont_data = BN_MONT_CTX_new();
+    if (group->mont_data == NULL){
+        goto err;
+    }
+
+    BN_MONT_CTX_init(group->mont_data);
+    BN_bin2bn(params+7*param_len, param_len, &(group->mont_data->RR));
+    group->mont_data->ri = 512;
+    group->mont_data->n0[0] = 14758798090332847183ULL;
```

Comment:
Search for TOBN()

Reply:
I assume you looked at BN_MONT_CTX_set and saw the
# if defined(OPENSSL_BN_ASM_MONT) && (BN_BITS2<=32) part